transit_odp/organisation/csv/service_codes.py

```diff
@@ -1,59 +1,251 @@
+                "line_number": service and service.service_number,
+                "dataset_id": file_attribute and file_attribute.revision.dataset_id,
+                "seasonal_status": seasonal_service
+                and seasonal_service.seasonal_status,
+                "seasonal_start": seasonal_service and seasonal_service.start,
+                "seasonal_end": seasonal_service and seasonal_service.end,
+                "staleness_status": staleness_status,
+                "effective_seasonal_start_date": seasonal_service
+                and seasonal_service.start - timedelta(days=42),
+                "effective_stale_date_end_date": file_attribute
+                and file_attribute.effective_stale_date_end_date,
+                "effective_stale_date_last_modified_date": file_attribute
+                and file_attribute.effective_stale_date_last_modified_date,
+                "last_modified_lt_effective_stale_date_otc": service
+                and (
+                    service.last_modified.date()
+                    < service.effective_stale_date_otc_effective_date
+                ),
+                "effective_stale_date_otc_effective_date": service
+                and (service.effective_stale_date_otc_effective_date),
+            }
         )
+
+    def get_otc_service_bods_data(self, organisation_id: int) -> None:
+        """
+        Compares an organisation's OTC Services dictionaries list with
+        TXCFileAttributes dictionaries list and the SeasonalService list
+        to determine which OTC Services require attention and which
+        doesn't ie. not live in BODS at all, or live but meeting new
+        Staleness conditions.
+        """
+        otc_map = get_all_otc_map(organisation_id)
+        txcfa_map = get_txc_map(organisation_id)
+        seasonal_service_map = get_seasonal_service_map(organisation_id)
+        service_code_exemption_map = get_service_code_exemption_map(organisation_id)
+        services_code = set(otc_map)
+        services_code.update(set(txcfa_map))
+        services_code = sorted(services_code)
+
+        for service_code in services_code:
+            service = otc_map.get(service_code)
+            file_attribute = txcfa_map.get(service_code)
+            seasonal_service = seasonal_service_map.get(service_code)
+            exemption = service_code_exemption_map.get(service_code)
+
+            staleness_status = "Not Stale"
+            if file_attribute is None:
+                if exemption or (
+                    seasonal_service and not seasonal_service.seasonal_status
+                ):
+                    require_attention = "No"
+                else:
+                    require_attention = "Yes"
+            elif service and requires_attention(service, file_attribute):
+                rad = requires_attention_data(service, file_attribute)
+                staleness_status = STALENESS_STATUS[rad.index(True)]
+                require_attention = "Yes"
```

Comment:
You still need to check for seasonality and for service code exemptions here. If it's exempted for either reason, it doesn't count as requiring attention.